stdpopsim/models.py

```diff
@@ -299,3 +299,67 @@ def __init__(self, NA, N1, N2, T, M12, M21):
             model=model,
             generation_time=1,
         )
+
+
+class DFE:
+    """
+    Class representing a DFE.
+
+    Instances of this class are constructed by dfe implementors, following the
+    :ref:`developer documentation <sec_development_def_model>`. To instead
+    obtain a pre-specified model as listed in the :ref:`sec_catalog`,
+    see :class:`Species.get_dfe`.
+
+    :ivar ~.mutation_types: a list of MutationTypes associated with the DFE.
+    :vartype ~.mutation_types: list
+    :ivar ~.proportions: a list of MutationTypes proportions
+    :vartype ~.proportions: list
+    :ivar ~.id: The unique identifier for this model. DFE IDs should be
+        short and memorable, and conform to the stdpopsim
+        ##TODO add the DFE naming conventions in the documentation
+        :ref:`naming conventions <sec_development_naming_conventions>`
+        for DFE models.
+    :vartype ~.id: str
+    :ivar ~.description: A short description of this model as it would be used in
+        written text, e.g., "Lognormal DFE". This should
+        describe the DFE itself and not contain author or year information.
+    :vartype ~.description: str
+    :ivar long_description: A concise, but detailed, summary of the DFE model.
+    :vartype long_description: str
+    :ivar citations: A list of :class:`Citation`, that describe the primary
+        reference(s) for the DFE model.
+    :vartype citations: list of :class:`Citation`
+    """
+
+    def __init__(
+        self,
+        *,
+        id,
+        description,
+        long_description,
+        mutation_types=None,
+        proportions=None,
+        citations=None,
+        # qc_model=None, # we may want to include qc of dfe in the future
+    ):
+        self.id = id
```

Comment:
Should we check that `mutation_types` and `proportions` are of the same length, and raise an error if not?

Reply:
Also check that proportions are sensible (eg not negative, don't sum to more than 1).

Reply:
and check that all the things in `mutation_types` are of class MutationType